src/Main.cpp

```diff
@@ -190,24 +192,48 @@ class CommMonitoringTask : public AbstractTask<4 * configMINIMAL_STACK_SIZE> {
     }
 };
 
-class InterlocTask : public AbstractTask<10 * configMINIMAL_STACK_SIZE> {
+class HardwareInterlocTask : public AbstractTask<10 * configMINIMAL_STACK_SIZE> {
   public:
-    InterlocTask(const char* taskName, UBaseType_t priority) : AbstractTask(taskName, priority) {}
+    HardwareInterlocTask(const char* taskName, UBaseType_t priority) :
+        AbstractTask(taskName, priority) {}
 
-    ~InterlocTask() override = default;
+    ~HardwareInterlocTask() override = default;
 
   private:
     void task() override { BSPContainer::getInterlocManager().startInterloc(); }
 };
 
+class SoftwareInterlocTask : public AbstractTask<10 * configMINIMAL_STACK_SIZE> {
```

Comment:
more of a "InterlocMessageHandlerTask " or there will be more processing here eventually?

Reply:
I was thinking of putting a process() for the Interloc class which could handle filtering and such